.github/workflows/ci.yaml

```diff
@@ -0,0 +1,88 @@
+name: Backup Restore CI
+
+permissions:
+  contents : read
+
+on:
+  workflow_call:
+  push:
+    branches:
+      - release/v5.0
+      - release/v4.0
+      - release/v3.0
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+      - '.gitignore'
+      - 'CODEOWNERS'
+      - 'LICENSE'
+  pull_request:
+    paths-ignore:
+    - 'docs/**'
+    - '*.md'
+    - '.gitignore'
+    - 'CODEOWNERS'
+    - 'LICENSE'
+jobs:
+  build:
+    strategy:
+      matrix:
+        os : [ 
+          ubuntu-latest
+          # org-rancher-arm64-containers
+          ]
+    name : CI
+    runs-on : ${{ matrix.os }}
+    outputs:
+      k3sversions: ${{ steps.support.outputs.k3sversions }}
+    steps:
+      - name : Checkout repository
+        uses : actions/checkout@v4
+        with :
+          fetch-depth : 0
+      - name: Set K3s support output
+        id : support
+        run: ./.github/workflows/scripts/supported-versions.sh | awk '{print "k3sversions="$0}' >> $GITHUB_OUTPUT
+      - name : CI
+        run : make ci
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: bro-build-artifacts-${{ matrix.os }}
+          path: ./dist/artifacts
+        
+  test:
+    name : integration-test
+    needs : [
+      build
+    ]
+    runs-on : ${{ matrix.os }}
+    strategy:
+      matrix:
+        os : [ 
+          ubuntu-latest
+          # org-rancher-arm64-containers,
+         ]
+        K3S_VERSION : ${{ fromJSON(needs.build.outputs.k3sversions) }}
+    steps:
+      - name : Checkout repository
+        uses : actions/checkout@v4
+      - name : Fetch build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: bro-build-artifacts-${{ matrix.os }}
+          path: ./dist/artifacts/
+      - name : Debug
```

Comment:
I use this too, but I assume we want to pull out the Debug parts once everything's working